Dockerfile

```diff
@@ -16,12 +16,16 @@ RUN \
   apt-get -y --force-yes --no-install-recommends install libfontconfig curl ca-certificates git jq && \
   curl https://grafanarel.s3.amazonaws.com/builds/grafana_${GRAFANA_VERSION}_amd64.deb > /tmp/grafana.deb && \
   dpkg -i /tmp/grafana.deb && \
-  rm -f /tmp/grafana.deb && \
-  curl -L https://github.com/tianon/gosu/releases/download/1.10/gosu-amd64 > /usr/sbin/gosu && \
-  chmod +x /usr/sbin/gosu && \
-  for plugin in $(curl -s https://grafana.net/api/plugins?orderBy=name | jq '.items[] | select(.internal=='false') | .slug' | tr -d '"'); do grafana-cli --pluginsDir "${GF_PLUGIN_DIR}" plugins install $plugin; done && \
+  rm -f /tmp/grafana.deb
+
+RUN  curl -L https://github.com/tianon/gosu/releases/download/1.10/gosu-amd64 > /usr/sbin/gosu && \
+  chmod +x /usr/sbin/gosu
+
+RUN \
+  for plugin in $(curl -s https://grafana.net/api/plugins?orderBy=name | jq '.items[] | select(.internal=='false') | .slug' | tr -d '"'); \
+    do rc=1;count=0; while [ $rc -ne 0 ] && [ $count -lt 5 ]; do grafana-cli --pluginsDir "${GF_PLUGIN_DIR}" plugins install $plugin;rc=$?;count=$(( count+1 )); done; done
```

Comment:
It would fail when the 'plugins install "$plugin"' would fail with any http error, connection timeout etc..

when it fails, the returncode is saved in $rc, and then the while conditions are reevaluated..
so the while only escapes when either the return code for the specific plugin is 0, or have been retried 6 times, then the for loop kicks back in with the next plugin..

Also, I'll fix the missing spaces after each `;` and also properly prevent globbing and word splitting with "$variables" (https://github.com/koalaman/shellcheck/wiki/SC2086)